### ISSUE TITLE
Negative/Oversize Queue Size

### ISSUE THREAD

**Issue by dchecks** [vterron/lemon#102]
https://github.com/vterron/lemon/blob/d60576bec2ad5d1d5043bcb3111dff1fcb58a8d6/methods.py#L554-L560

This will cause a negative size if .get with a timeout is called on an empty queue as a queue.Empty exception is raised.

Calling .put will artificially inflate the size due to a queue.Full exception

Without explicitly locking (as my situation doesn't require it) the following solution works:
```
    def put(self, *args, **kwargs):
        super(Queue, self).put(*args, **kwargs)
        self.size.increment(1)

    def get(self, *args, **kwargs):
        item = super(Queue, self).get(*args, **kwargs)
        self.size.increment(-1)
        return item
```

Also, thank you for this solution!

**Comment by vterron** [comment]
Good catch! Would you be willing to send a pull request with the change?

For get() another possible approach would be a try-finally.

**Comment by vterron** [comment]
Oops, my mistake. You're right, try-finally doesn't work here! As you pointed out, we only want to decrement the counter if  _no_ exception is raised. Sorry for the misleading comment!

**Comment by vterron** [comment]
Fixed in #103.
